frontend/packages/console-app/src/components/cloud-vm-console/cloud-vm-console.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { connectToFlags, FlagsObject } from '@console/internal/reducers/features';
+import ConnectedVMConsole from '@console/kubevirt-plugin/src/components/connected-vm-console/connected-vm-console';
+import { ConsoleType } from '@console/kubevirt-plugin/src/constants/vm/console-type';
+
+const CloudVMConsole: React.FC<CloudVMConsoleProps> = ({ location, flags }) => {
+  const params = new URLSearchParams(location.search);
+  const type = ConsoleType.fromString(params.get('console'));
+  const namespace = params.get('namespace');
+  const name = params.get('name');
+
+  return (
+    <ConnectedVMConsole
+      isKubevirt={flags.KUBEVIRT}
+      type={type === ConsoleType.SERIAL ? ConsoleType.SERIAL : ConsoleType.VNC}
```

Comment:
Why not simply use value pass-through?

```suggestion
      type={type}
```

The value should be already coerced via `ConsoleType.fromString` logic.